### PR TITLE
A few forward AD formulas

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #77421

It includes all-time favourites like:
- `put`
- `nn.functional.embedding`
- `prelu`
- `nn.functional.bilinear`
- `nn.functional.rrelu`
- `nn.functional.logsigmoid`

### CODE REVIEW

File: aten/src/ATen/native/Activation.cpp [hunk 1]
```diff
@@ -588,14 +588,13 @@ Tensor rrelu_with_noise_backward(
...
+  if (training) {
+    return noise * grad_output;
   } else {
-    auto negative = (lower_tensor + upper_tensor) / 2;
-    Scalar negative_slope = negative.item();
-    return at::leaky_relu_backward(grad_output, self_or_result, negative_slope, is_result);
+    auto l = lower.toDouble();
+    auto u = upper.toDouble();
+    auto mid = (l + u) / 2.;
+    return at::leaky_relu_backward(grad_output, self_or_result, mid, is_result);
```
Comment (Lezcano) [inline aten/src/ATen/native/Activation.cpp:597]: This is just a simplification of the previous formula (which was incorrect when `lower > upper` I believe) that avoids using and inspecting the elements of tensors.

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -13240,6 +13240,13 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
         skips=(
             # lambda impl
             DecorateInfo(unittest.expectedFailure, 'TestNormalizeOperators', 'test_normalize_operator_exhaustive'),
+            # In-place operations do not play well with forward AD
+            # FIXME it should be fixable, as auto_elementwise works with in-place just fine, and
+            # this one is somewhat similar
+            # INTERNAL ASSERT FAILED Cannot set a va lue of a forward grad if it already exists.
```
Comment (soulitzer) [inline torch/testing/_internal/common_methods_invocations.py:13246]: Could you file an issue for this?

**Review by soulitzer — APPROVED** [review]
Thanks!

File: torch/testing/_internal/common_methods_invocations.py (hunk 2 above)
Comment (Lezcano) [reply torch/testing/_internal/common_methods_invocations.py]: https://github.com/pytorch/pytorch/issues/77447

**Pull request closed without merging**